device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/twin/Twin.java

```diff
@@ -158,7 +155,12 @@ public Twin(TwinCollection desiredProperty, TwinCollection reportedProperty)
      */
     public JsonElement toJsonElement()
     {
-        Gson gson = new GsonBuilder().disableHtmlEscaping().serializeNulls().create();
+        Gson gson = new GsonBuilder()
+                .disableHtmlEscaping()
+                .setObjectToNumberStrategy(ToNumberPolicy.LONG_OR_DOUBLE)
```

Comment:
Setting this in our gson object makes it so gson does not convert our ints into doubles